src/Interpreters/InterpreterDropNamedCollectionQuery.cpp

```diff
@@ -32,7 +32,8 @@ BlockIO InterpreterDropNamedCollectionQuery::execute()
     }
 
     NamedCollectionFactory::instance().removeFromSQL(query);
-    CurrentMetrics::sub(CurrentMetrics::NamedCollection);
+    if (CurrentMetrics::get(CurrentMetrics::NamedCollection) > 0)
+        CurrentMetrics::sub(CurrentMetrics::NamedCollection);
```

Comment:
I would suggest throwing a logical error, since somehow we dropped a named collection that wasn't added to the metrics. I feel like it needs a separate fix, to find the place where we forgot to add this metric.